Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -179,14 +221,24 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
                 for (int iy=0; iy<=depos_order; iy++){
                     for (int ix=0; ix<=depos_order; ix++){
                         amrex::Gpu::Atomic::AddNoRet(
-                            &rho_arr(lo.x+i+ix, lo.y+j+iy, lo.z+k+iz),
+                            &rho_tmp(lo.x+i+ix, lo.y+j+iy, lo.z+k+iz),
                             sx[ix]*sy[iy]*sz[iz]*wq);
                     }
                 }
             }
 #endif
         }
-        );
+
+#if defined(AMREX_USE_GPU)
+        __syncthreads();
+
+        for (int i = threadIdx.x; i < rho_box.numPts(); i += blockDim.x) {
+          amrex::Real r = vs[i];
+          amrex::Gpu::Atomic::AddNoRet(rho_arr.dataPtr()+i, r);
```

Comment:
Just for my own understanding, what is the intended use for `rho_arr` and `rho_temp`? I see that earlier, charge is deposited into rho_temp, but I don't see where values are being transferred to `rho_arr` or `vs`.

Reply:
I've got pointers aliasing the same memory. `rho_arr` is the durable data in GPU global memory. `rho_tmp` is the space in the GPU's fast shared memory in which the values get accumulated, and `vs` is a pointer alias to that same memory, but with the `volatile` qualifier, to ensure that the final values of the accumulation get read, following the `__syncthreads()` call, rather than a value that the thread may have cached.